### PR TITLE
Apply ACLs on API

### PR DESCRIPTION
**Pull request by mmadariaga**
Check admin ACLs on API

#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX)

### CODE REVIEW

File: library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php [hunk 1]
```diff
@@ -64,6 +67,60 @@ public function getRoles()
         ];
     }
 
+    /**
+     * @param string $fqdn
+     * @param string $reqMethod
+     * @return bool
+     * @throws \Exception
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php]: Does not throw exceptions anymore + this comment is not necessary at all (DRY)

File: library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php [hunk 2]
```diff
@@ -64,6 +67,60 @@ public function getRoles()
...
+            'POST' => 'create',
+            'GET' => 'read',
+            'PUT' => 'update',
+            'PATCH' => 'update',
+            'DELETE' => 'delete'
+        ];
+
+        try {
+            Assertion::choice(
+                $reqMethod,
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php]: Preventive strtoupper?

File: library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php [hunk 3]
```diff
@@ -64,6 +67,60 @@ public function getRoles()
...
+                    ->getPublicEntity()
+                    ->getFqdn();
+
+                if ($entityFqdn === $fqdn) {
+                    return true;
+                }
+            }
+        } catch (\Exception $e) {
+            return false;
+        }
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php]: finally return false

File: web/rest/client/app/config/api/raw/provider.yml [hunk 4]
```diff
@@ -183,7 +183,7 @@ Ivoz\Provider\Domain\Model\CallForwardSetting\CallForwardSetting:
         tags:
         - My
   attributes:
-    access_control: '("ROLE_COMPANY_ADMIN" in roles) or ("ROLE_COMPANY_USER" in roles)'
+    access_control: '("ROLE_COMPANY_ADMIN" in roles && user.hasAccessPrivileges(_api_resource_class, request.getMethod())) or ("ROLE_COMPANY_USER" in roles && user.hasAccessPrivileges(_api_resource_class, request.getMethod()))'
```
Comment (mmadariaga) [inline web/rest/client/app/config/api/raw/provider.yml]: ROLE_COMPANY_USER has no ACLs

**Comment by ironArt3mis** [comment]
Rebase completed.

File: library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php [hunk 5]
```diff
@@ -64,6 +67,56 @@ public function getRoles()
...
+        ];
+
+        try {
+            Assertion::choice(
+                strtoupper($reqMethod),
+                array_keys($accessMethods),
+                'Request method "%s" is not an element of the valid values: %s'
+            );
+
+            $accessMethod = $accessMethods[$reqMethod];
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php]: strtoupper too

File: library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php [hunk 6]
```diff
@@ -64,6 +67,56 @@ public function getRoles()
         ];
     }
 
+    public function hasAccessPrivileges(string $fqdn, string $reqMethod): bool
+    {
+        $accessMethods = [
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/Administrator/AdministratorSecurityTrait.php]: if not restricted return true

**Pull request merged**